platforms/core-configuration/model-core/src/test/groovy/org/gradle/api/internal/provider/CollectionPropertySpec.groovy

```diff
@@ -1362,76 +1365,73 @@ The value of this property is derived from: <source>""")
         property.explicit
     }
 
-    def "appending to an undefined property is undefined-safe"() {
+    def "test '#label' vs undefined-safety"(String label) {
```

Comment:
Need to resort to a data-driven test case to be sure I was testing `add` and `append` consistently.

Deleted the corresponding method-based tests.

Reply:
> Need to resort to a data-driven test case to be sure I was testing `add` and `append` consistently.
> 
> Deleted the corresponding method-based tests.